test/unit/core.js

```diff
@@ -1709,3 +1709,45 @@ QUnit.test( "Iterability of jQuery objects (gh-1693)", function( assert ) {
 		assert.ok( true, "The browser doesn't support Symbols" );
 	}
 } );
+
+QUnit[ jQuery.Deferred ? "test" : "skip" ]( "jQuery.readyException (original)", function( assert ) {
+	assert.expect( 1 );
+
+	var message,
+		origWindowSetTimeout = window.setTimeout;
+
+	window.setTimeout = function( fn ) {
```

Comment:
Not necessary to use sandbox i think, sandbox usually needed to clean up some scope of spies/stubs, you probably could just use 

`sinon.stub(window, 'setTimeout', myOwnThing);`